File: 03 C C++ Python R and Mojo review/03-3 R Language Review/01 Pointers/05.R
# R equivalent for iterating through an array-like structure

arr <- c(12, 24, 36, 48, 60)

# Position one (first element)
cat("Position one:", arr[1], "\n")

# Iterate over the array and simulate printing memory address
for (num in arr) {
    cat(num, "\t", pryr::address(num), "\n")
    # pryr::address() returns a simulated memory address in R
}

File: 03 C C++ Python R and Mojo review/03-2 Python Review/01 Pointers/05.py
# Python equivalent for iterating through an array-like structure

arr = [12, 24, 36, 48, 60]

# Position one (first element)
print(f"Position one: {arr[0]}")

# Iterate over the array and simulate printing memory address
for num in arr:
    print(f"{num}\t{id(num)}")
    # id() gives a unique identifier for the object, similar to a memory address